### PR TITLE
Fix crash in Feeds and Starter Packs

### PR DESCRIPTION
**Pull request by gaearon**
See inline comments.

## Test Plan

Add `throw Error('no')` [here](https://github.com/bluesky-social/social-app/blob/873d91d4664403577fff0438bfc81304f1fafe5b/src/state/queries/feed.ts#L544), then add it [here](https://github.com/bluesky-social/social-app/blob/873d91d4664403577fff0438bfc81304f1fafe5b/src/state/queries/feed.ts#L531). In both cases the Feeds page should be able to recover, and show the rest instead. Similarly start pack creation should recover in both cases.

### CODE REVIEW

File: src/screens/StarterPack/Wizard/StepFeeds.tsx [hunk 1]
```diff
@@ -41,13 +41,9 @@ export function StepFeeds({moderationOpts}: {moderationOpts: ModerationOpts}) {
...
   })
   const popularFeeds = popularFeedsPages?.pages.flatMap(p => p.feeds) ?? []
-
-  const suggestedFeeds =
-    savedFeeds.length === 0
-      ? popularFeeds
-      : savedFeeds.concat(
-          popularFeeds.filter(f => !savedFeeds.some(sf => sf.uri === f.uri)),
-        )
+  const suggestedFeeds = savedFeeds.concat(
```
Comment (gaearon) [inline src/screens/StarterPack/Wizard/StepFeeds.tsx:44]: Not essential to the fix, but this wasn't doing anything and made the code more complicated.

File: src/state/queries/feed.ts [hunk 2]
```diff
@@ -556,28 +557,39 @@ export function useSavedFeeds() {
...
+        if (savedItem.type === 'timeline') {
+          result.push({
             type: 'timeline',
-            config: s,
+            config: savedItem,
             view: undefined,
+          })
+        } else if (savedItem.type === 'feed') {
+          const resolvedFeed = resolvedFeeds.get(savedItem.value)
+          if (resolvedFeed) {
```
Comment (gaearon) [inline src/state/queries/feed.ts:570]: This is the fix.

File: src/state/queries/feed.ts [hunk 3]
```diff
@@ -556,28 +557,39 @@ export function useSavedFeeds() {
...
+            result.push({
+              type: 'feed',
+              config: savedItem,
+              view: resolvedFeed,
+            })
+          }
+        } else if (savedItem.type === 'list') {
+          const resolvedList = resolvedLists.get(savedItem.value)
+          if (resolvedList) {
```
Comment (gaearon) [inline src/state/queries/feed.ts:579]: And this.

File: src/state/queries/feed.ts [hunk 4]
```diff
@@ -556,28 +557,39 @@ export function useSavedFeeds() {
...
-
-        return {
-          type: s.type,
-          config: s,
-          view:
-            s.type === 'feed'
-              ? resolvedFeeds.get(s.value)
-              : resolvedLists.get(s.value),
-        }
-      }) as SavedFeedItem[]
```
Comment (gaearon) [inline src/state/queries/feed.ts:576]: This was covering up the mistake.

**Review by estrattonbailey — APPROVED** [review]

**Pull request merged**